tools/install_opte.sh

```diff
@@ -52,36 +53,44 @@ function sha_from_url {
     curl -L "$SHA_URL" 2> /dev/null | cut -d ' ' -f 1
 }
 
-OPTE_P5P_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G0MRWX9Y0X46HBEJBW245DJY/PM097Agvf89uKmVRZ890z6saoeLp6RCcVsbYRa5PDv9DnLDT/01G0MRX6GMBV34CNANABXZXX25/01G0MSFZZWPFEQBW7JRS7ST99G/opte-0.1.58.p5p"
-OPTE_P5P_SHA_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G0MRWX9Y0X46HBEJBW245DJY/PM097Agvf89uKmVRZ890z6saoeLp6RCcVsbYRa5PDv9DnLDT/01G0MRX6GMBV34CNANABXZXX25/01G0MSG01CGP6TH9THNY39G88Z/opte-0.1.58.p5p.sha256"
-OPTE_P5P_REPO_PATH="$OUT_DIR/$(basename "$OPTE_P5P_URL")"
-XDE_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G0DM53XR4E008D6ET5T8DXP6/wBWo0Jsg1AG19toIyAY23xAWhzmuNKmAsF6tL18ypZODNuHK/01G0DM5DMQHF5B89VGHZ05Z4E0/01G0DMHNYQ1NS7DBX8VG3JPAP0/xde"
-XDE_SHA_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G0DM53XR4E008D6ET5T8DXP6/wBWo0Jsg1AG19toIyAY23xAWhzmuNKmAsF6tL18ypZODNuHK/01G0DM5DMQHF5B89VGHZ05Z4E0/01G0DMHP47353961S3ETXBSD2T/xde.sha256"
+# Add a pkg repo by path. This will also run `pkg update`, handling the case
+# gracefully where there are no updates required
+function add_pkg_repo_and_update {
+    local REPO_PATH="$1"
+    pkg set-publisher -p "$REPO_PATH" --search-first
+    RC=0
+    pkg update || RC=$?;
+    if [[ "$RC" -eq 0 ]] || [[ "$RC" -eq 4 ]]; then
+        return 0
+    else
+        return "$RC"
+    fi
+}
 
-download_and_check_sha "$OPTE_P5P_URL" "$(sha_from_url "$OPTE_P5P_SHA_URL")"
-XDE_SHA="$(sha_from_url "$XDE_SHA_URL")"
-download_and_check_sha "$XDE_URL" "$XDE_SHA"
+# The `helios-netdev` provides the XDE kernel driver and the `opteadm` userland
+# tool for interacting with it.
+HELIOS_NETDEV_REPO_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G11AT7E4XV9J1J54GE2YDJT6/CB4WF4BVgnbvf5NI573z9osAV2LNIKogPtWJ5sfW2cNxUYQO/01G11ATFVTWAC2HSNV148PQ4ER/01G11B5MPQRBX3Q5EF45YDAW6Q/opte-0.1.60.p5p"
+HELIOS_NETDEV_REPO_SHA_URL="https://buildomat.eng.oxide.computer/wg/0/artefact/01G11AT7E4XV9J1J54GE2YDJT6/CB4WF4BVgnbvf5NI573z9osAV2LNIKogPtWJ5sfW2cNxUYQO/01G11ATFVTWAC2HSNV148PQ4ER/01G11B5MR60H4N13NJKGWEEA69/opte-0.1.60.p5p.sha256"
+HELIOS_NETDEV_REPO_PATH="$XDE_DIR/$(basename "$HELIOS_NETDEV_REPO_URL")"
```

Comment:
So I just want to make sure that this matches up with the specific opte/opte-ioctl crate revisions you pull, correct? Just something to keep in mind because if the API spec in OPTE bumps, and we update the crate versions but not this, then we could get in a situation where the sled agent cannot talk to OPTE. Though it should hopefully have a fairly obvious error about API version mismatch.

Reply:
Yeah I was worried about this. For now, I'm just planning to keep these in sync manually, as much of a pain as that may be. I need to ask Josh if there's a way to refer to these that only relies on the version number, not the gobbledegook of the Buildomat runner information.

Reply:
Well it means we have to figure out what "version number" is, and how we map that between OPTE's API version number and whatever this thing is you are proposing. But that's something we can figure out separate from this PR.

Reply:
Buildomat now supports making build artifacts public. They're exposed by a few names we can pick, but the version is ultimately the commit SHA. One of the names is a "series", which could conceivably be or include a version number that's the same between the OPTE P5P repo and the one providing the kernel bits that `xde` works with. I've started experimenting with that [here](https://github.com/oxidecomputer/opte/pull/130), but perhaps we can decide on a versioning scheme that keeps those two repositories in sync. cc @rcgoodfellow

Reply:
I've put this into the latest changes, where both the OPTE P5P package (`opteadm` and the `xde` driver) and the package implementing the kernel bits that `xde` requires all pulled from a public file published by buildomat. These are referred to by commit, so it's still possible that the driver and kernel bits get out of sync, but it's easier to refer to a specific version by its SHA.